core/src/main/java/hudson/model/AutoCompletionCandidates.java

```diff
@@ -118,16 +119,30 @@ public void generateResponse(StaplerRequest req, StaplerResponse rsp, Object o)
             @Override
             public void onItem(Item i) {
                 String n = contextualNameOf(i);
-                if ((n.startsWith(value) || value.startsWith(n))
+                
+                //Check user's setting on whether to do case sensitive comparison, configured in user -> configure
+                //This is the same setting that is used by the global search field, should be consistent throughout
+                //the whole application.
+                boolean caseInsensitive = UserSearchProperty.isCaseInsensitive(); 
```

Comment:
🐛  UserSearchProperty is not about jobs at all, so such check is confusing.
I would rather ALWAYS perform case-insensitive for items.